File: Python/FindIndexOfFirstOccurence.py
def strStr(haystack,needle):
    lenNeedle = len(needle)
    lenHaystack = len(haystack)

    if lenNeedle>lenHaystack:
        return -1
    
    if needle == haystack:
        return 0
    

    for i in range(lenHaystack-lenNeedle+1):
        substring = haystack[i:i+lenNeedle]
        if substring == needle:
            return i
        
    return -1

File: Javascript/FindIndexOfFirstOccurence.js
var strStr = function (haystack, needle) {
  if (haystack.length < needle.length) {
    return -1;
  }

  if (haystack === needle) {
    return 0;
  }

  for (let i = 0; i < haystack.length - needle.length + 1; i++) {
    let subString = haystack.slice(i, i + needle.length);
    if (subString === needle) {
      return i;
    }
  }

  return -1;
};
